docs/dev/Kibana-Reporting-Design-Proposal.md

```diff
@@ -20,43 +21,111 @@ Kibana Reports for Open Distro allows ‘Report Owner’ (engineers, including b
-        * As a user, when I click a particular report in the list and it belongs to the alert based reports category, I should be able to edit and save, change the report name, share if not already shared, choose another alert, stop/pause/resume a report, archive stopped reports, delete report, change the file format of the report, sharing coordinates, sharing options.
-        * As a user, I should be able to see a list of reports archived for the past 100 days.
-    * For Received Reports
-        * As a user, I should see report name, frequency, status will always be ‘shared’, start date, end date, name and role of the person who shared the report
-        * As a user, when I click a particular report in the list, I should be able to view the report in a browser using a URL. 
-        * As a user, I should be able to archive reports. 
-            * Delete option should not be there as it might lead to complications such as one wherein the user who receives the reports deletes them and says that they never received the reports
-        * As a user, I should be able to see a list of reports archived for the past 100 days. NOTE: Reports will be stored as hyperlinks/urls and NOT as actual reports. If the data is available at the time when the url is clicked, report will be created and can be viewed in a browser and can be downloaded.
-
-### Functional 
-
+1. As a user, I want to see a 'Share >' 'As a report' on the share menu, and a 'Download > ' 'PDF, CSV, PNG' contextually button in the  Dashboard, Visualizations, and Discover Tabs.
+1. As a user, I should be able to export and share reports only if I have the permissions to do so. Else, an error should be thrown.
+1. As a user, when I click ‘Share > As a report’ I should be able to configure, save, export and share the report.
+1. As a user, when I click ‘Share > As a report’, the system should allow me to share the dashboard or saved search or visualizations that I am currently working on as a report.
+
+    Here is the matrix showing various panels and what kind of export option should be available.        
+    Kibana panel | PDF           | PNG           |Hyperlink (HTML) | CSV/Excel
+    ------------ | ------------- | ------------- | ------------- | -------------
+    Dashboards | Yes | Yes | Yes | No
+    Visualizations | Yes |  Yes |Yes | No
+    Discover (saved search) | No | No | Yes | Yes
+
+### In-context reports
+1. As a user, I can download a PDF, PNG, CSV, XLS file in-context from any available panel (e.g. Dashboards, visualizations, saved searches, etc).
+1. As a user, any time I download a report from any available panel (e.g. Dashboards, visualizations, saved searches, etc), a report will be auto-generated in the reports list.
+1. As a user, I can create a report definition from the share menu of any available panel (e.g. Dashboards, visualizations, saved searches, etc).
```

Comment:
From doc- When `download` is clicked from the share menu, it will direct to the `Create Report` page with settings pre-populated contextually

Reply:
This was before we decided to split `Download` from `Share`. This current user story reflects the change in thinking. The flow is `Share -> Create report definition`, or `Download -> PDF report | PNG report | CSV report | XLS report`